docs/deploy-lab.md

```diff
@@ -0,0 +1,355 @@
+
+## Next steps
+
+Congratulations! You've now seen how to use GitHub Actions to enable blue-green deployments. You started by configuring the resources on Azure, configuring the repository with the settings necessary for the workflow, and created the workflow itself. You finished by testing the workflow by making a successful and unsuccessful change to the application.
+
+From here, you can continue to explore and grow your knowledge of DevOps. Here are some suggestions:
+
+- [About GitHub workflows](https://docs.github.com/actions/using-workflows/about-workflows)
+- [Continuous integration](https://docs.github.com/actions/automating-builds-and-tests/about-continuous-integration)
+- [Deploying with GitHub Actions](https://docs.github.com/actions/deployment/about-deployments/about-continuous-deployment)
+- [Monitoring and troubleshooting GitHub Actions](https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/about-monitoring-and-troubleshooting)
+
+## Complete workflow file
+
+```yaml
+name: Azure Container Apps Deploy
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Build and deploy Container App
+        uses: azure/container-apps-deploy-action@v0
+        with:
+          appSourcePath: ${{ github.workspace }}/load_balancer/ # Location of Dockerfile
+          acrName: ${{ secrets.ACR_NAME }} # Name of Azure Container Registry
+          containerAppName: ${{ secrets.CONTAINER_APP_NAME }} # Name of Azure Container App
+          resourceGroup: ${{ secrets.RESOURCE_GROUP }} # Name of Azure Resource Group
+  
+  test-deployment:
+    needs: build-deploy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: Get new container name
+        run: |
+          # Enable extension installation
+          az config set extension.use_dynamic_install=yes_without_prompt
+          # Get the last deployed revision name
+          REVISION_NAME=`az containerapp revision list -n ${{ secrets.CONTAINER_APP_NAME }} -g ${{ secrets.RESOURCE_GROUP }} --query "[].name" -o tsv | tail -1`
+          # Get the last deployed revision's fqdn
+          REVISION_FQDN=`az containerapp revision show -n ${{ secrets.CONTAINER_APP_NAME }} -g ${{ secrets.RESOURCE_GROUP }} --revision "$REVISION_NAME" --query properties.fqdn -o tsv`
+          # Store values in env vars
+          echo "REVISION_NAME=$REVISION_NAME" >> $GITHUB_ENV
+          echo "REVISION_FQDN=$REVISION_FQDN" >> $GITHUB_ENV
+      - name: Test deployment
+        id: test-deployment
+        uses: jtalk/url-health-check-action@v3 # Marketplace action to touch the endpoint
```

Comment:
Question that just occurred to me:  Does this require the health check endpoint to be public? If I'm releasing, say, a backend service, would it require a different approach?

Reply:
To be clear, not a change request - asking purely for my learning